quickwit/quickwit-metastore/src/metastore/file_backed_metastore/mod.rs

```diff
@@ -457,23 +456,17 @@ impl Metastore for FileBackedMetastore {
     /// -------------------------------------------------------------------------------
     /// Read-only accessors
 
-    async fn list_splits(
-        &self,
-        index_id: &str,
-        state: SplitState,
-        time_range_opt: Option<Range<i64>>,
-        tags: Option<TagFilterAst>,
-    ) -> MetastoreResult<Vec<Split>> {
-        self.read(index_id, |index| {
-            index.list_splits(state, time_range_opt, tags, None)
+    async fn list_splits<'a>(&self, filter: ListSplitsQuery<'a>) -> MetastoreResult<Vec<Split>> {
+        self.read(filter.index, |index| {
+            if filter.is_default() {
+                index.list_all_splits()
+            } else {
+                index.list_splits(filter)
```

Comment:
Why can't we always use `index.list_splits(filter)`? In the first case, the query is simply a nop.